eng/common/pipelines/templates/steps/sparse-checkout.yml

```diff
@@ -67,16 +67,21 @@ steps:
             New-Item $dir -ItemType Directory -Force | Out-Null
             Push-Location $dir
 
+            $ghToken = ""
+            if ($repository.Name.EndsWith("-pr")) {
+              $ghToken = "$(azuresdk-github-pat)@"
```

Comment:
I don't like this approach so I'm investigating if I can do this with a git config header.